File: backend/paylink/src/main/java/edu/cit/sevilla/paylink/common/ApiError.java
package edu.cit.sevilla.paylink.common;

import java.time.LocalDateTime;
import java.util.Map;

public record ApiError(
        LocalDateTime timestamp,
        int status,
        String error,
        String message,
        Map<String, String> fieldErrors
) {
    public ApiError(int status, String error, String message) {
        this(LocalDateTime.now(), status, error, message, null);
    }

    public ApiError(int status, String error, String message, Map<String, String> fieldErrors) {
        this(LocalDateTime.now(), status, error, message, fieldErrors);
    }
}

File: mobile/app/src/main/java/edu/cit/sevilla/paylink/mobile/core/network/ApiError.kt
package edu.cit.sevilla.paylink.mobile.core.network

data class ApiError(
    val status: Int? = null,
    val error: String? = null,
    val message: String? = null,
    val details: Map<String, String>? = null,
    val path: String? = null,
    val timestamp: String? = null,
)
